pubsub/kafka/kafka.go

```diff
@@ -49,20 +52,39 @@ type kafkaMetadata struct {
 }
 
 type consumer struct {
+	k        *Kafka
```

Comment:
should only `bo` and `logger` be passed on this struct? Why is Kafka struct a pass by ref?

Reply:
Good catch, let me fix that.